### ISSUE TITLE
template validation

### ISSUE THREAD

**Issue by driegel** [elastic/puppet-elasticsearch#547]
This request is a feature enhancement. Template installation may fail as described in issue #540 .

Installing a template like this fails if elasticsearch does'nt use http.bind = ["127.0.0.1", ...]
```
elasticsearch::template{'template-name':
        content=> template('template-file.erb'),
}
```
The problem is, that puppet marks the template as installed in some internal state and won't fix it in a second run with proper host settings. (unless the template checksum has changed because it's content has changed)

To fix it, we have to use ensure => 'absent' first before the template will be installed again.
```
elasticsearch::template{'template-name':
        content=> template('template-file.erb'),
        ensure => 'absent'
        #ensure => 'present',
        host => 'localhost6',
        #host => 'ip6-localhost',
}
```

My suggestion would be to always make an 'GET _template/template-name' request to validate if a template was properly installed in elasticsearch.

**Comment by electrical** [comment]
I'm planning to write a custom type & provider for this part as the current implementation is indeed fragile and fails in so many different ways.
Now i have more experience with it after writing the plugin type & provider it should be easier to do this for the other components.

**Comment by electrical** [comment]
Tracking it via https://github.com/elastic/puppet-elasticsearch/issues/548

**Comment by tylerjl** [comment]
Fully-featured native type and provider for `elasticsearch::template` implemented in #663, should be in the next release. 👍
